imap_processing/ialirt/utils/create_xarray.py

```diff
@@ -0,0 +1,96 @@
+"""Creates xarray based on structure of queried DynamoDB."""
+
+from collections import defaultdict
+
+import numpy as np
+import xarray as xr
+
+from imap_processing.cdf.imap_cdf_manager import ImapCdfAttributes
+
+
+def create_xarray_from_records(records: list[dict]) -> xr.Dataset:
+    """
+    Create dataset from a list of records.
+
+    Parameters
+    ----------
+    records : list of dict
+       Output of querying DynamoDB.
+
+    Returns
+    -------
+    dataset : xarray.Dataset
+        Dataset in standard format.
+    """
+    cdf_manager = ImapCdfAttributes()
+    cdf_manager.add_instrument_global_attrs("ialirt")
+    cdf_manager.add_instrument_variable_attrs("ialirt", "l1")
+
+    instrument_prefixes = ("swe", "hit", "mag", "codicelo", "codicehi", "swapi")
+    instrument_keys: set[str] = set()
+
+    for record in records:
+        instrument_keys.update(
+            key for key in record if key.startswith(instrument_prefixes)
+        )
+
+    # Convert to columns, add fillvals, and associate with datatype.
+    data_dict = defaultdict(list)
+    for record in records:
+        for key in instrument_keys:
```

Comment:
Shouldn't you be looping over everything within the records. What happens if you have records that have content that you miss here, that would silently be dropped.